google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -200,15 +200,18 @@ def _process_requests(
                 exc = AcknowledgeError(AcknowledgeStatus.FAILED_PRECONDITION, info=None)
             else:
                 exc = AcknowledgeError(AcknowledgeStatus.OTHER, str(error_status))
-            future = ack_reqs_dict[ack_id].future
-            future.set_exception(exc)
-            requests_completed.append(ack_reqs_dict[ack_id])
+            if ack_id in ack_reqs_dict:
```

Comment:
This if is unnecessary b/c `acq_id` comes directly from iterating over `ack_reqs_dict`.

If mymy is giving you trouble, you can simply "assert" a statement. It'll then respect that.